### PR TITLE
Update grafusion

### PR DESCRIPTION
**Pull request by Umaaz**
**What is this feature?**

This updates grafana 10.2.3 to include the customizations for grafusion.

### CODE REVIEW

File: public/app/intergral/OpspilotDataLinkButton.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+// import { Field, LinkModel } from '@grafana/data';
```
Comment (Umaaz) [inline public/app/intergral/OpspilotDataLinkButton.tsx]:
```suggestion
```

File: public/app/intergral/OpspilotDataLinkButton.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+
+import { ButtonProps, Button } from '@grafana/ui';
```
Comment (Umaaz) [inline public/app/intergral/OpspilotDataLinkButton.tsx]:
```suggestion
import { ButtonProps, Button } from '@grafana/ui';
```

File: public/app/core/components/AppChrome/NavToolbar/NavToolbar.tsx [hunk 3]
```diff
@@ -58,21 +58,21 @@ export function NavToolbar({
...
+      {/*      state.megaMenu === 'closed' || (state.megaMenu === 'docked' && isTooSmallForDockedMenu)*/}
+      {/*        ? t('navigation.toolbar.open-menu', 'Open menu')*/}
+      {/*        : t('navigation.toolbar.close-menu', 'Close menu')*/}
+      {/*    }*/}
+      {/*    tooltipPlacement="bottom"*/}
+      {/*    size="xl"*/}
+      {/*    onClick={onToggleMegaMenu}*/}
+      {/*    data-testid={Components.NavBar.Toggle.button}*/}
+      {/*  />*/}
+      {/*</div>*/}
```
Comment (Umaaz) [inline public/app/core/components/AppChrome/NavToolbar/NavToolbar.tsx]:
```suggestion
```

File: public/app/intergral/OpspilotDataLinkButton.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { ButtonProps, Button } from '@grafana/ui';
```
Comment (Umaaz) [inline public/app/intergral/OpspilotDataLinkButton.tsx:3]:
```suggestion

import { ButtonProps, Button } from '@grafana/ui';
```

**Comment by Umaaz** [comment]
I have made some changes to re add some code that was removed. The functionality can be achieved by disabling the feature in the config.

File: conf/defaults.ini [hunk 5]
```diff
@@ -1357,6 +1357,10 @@ news_feed_enabled = true
 # Set the number of data source queries that can be executed concurrently in mixed queries. Default is the number of CPUs.
 concurrent_query_limit =
 
+[kiosk]
```
Comment (dhodgsonintergral) [inline conf/defaults.ini:1360]: Adds an item for the grafana config for configuring kiosk mode

File: packages/grafana-data/src/types/config.ts [hunk 6]
```diff
@@ -170,6 +170,7 @@ export interface GrafanaConfig {
   alertingMinInterval: number;
   authProxyEnabled: boolean;
   exploreEnabled: boolean;
+  kioskMode: string;
```
Comment (dhodgsonintergral) [inline packages/grafana-data/src/types/config.ts:173]: variable for kiosk mode

File: packages/grafana-runtime/src/config.ts [hunk 7]
```diff
@@ -63,6 +63,7 @@ export class GrafanaBootConfig implements GrafanaConfig {
   angularSupportEnabled = false;
   authProxyEnabled = false;
   exploreEnabled = false;
+  kioskMode = 'off';
```
Comment (dhodgsonintergral) [inline packages/grafana-runtime/src/config.ts:66]: setting the default for kiosk mode

File: pkg/api/dtos/frontend_settings.go [hunk 8]
```diff
@@ -146,15 +146,16 @@ type FrontendSettingsDTO struct {
...
-	HelpEnabled         bool `json:"helpEnabled"`
-	ProfileEnabled      bool `json:"profileEnabled"`
-	NewsFeedEnabled     bool `json:"newsFeedEnabled"`
-	QueryHistoryEnabled bool `json:"queryHistoryEnabled"`
+	VerifyEmailEnabled  bool   `json:"verifyEmailEnabled"`
+	SigV4AuthEnabled    bool   `json:"sigV4AuthEnabled"`
+	AzureAuthEnabled    bool   `json:"azureAuthEnabled"`
+	RbacEnabled         bool   `json:"rbacEnabled"`
+	ExploreEnabled      bool   `json:"exploreEnabled"`
+	KioskMode           string `json:"kioskMode"`
```
Comment (dhodgsonintergral) [inline pkg/api/dtos/frontend_settings.go:154]: linking the json kioskmode to a go variable

File: pkg/api/frontendsettings.go [hunk 9]
```diff
@@ -129,6 +129,7 @@ func (hs *HTTPServer) getFrontendSettings(c *contextmodel.ReqContext) (*dtos.Fro
 		ProfileEnabled:                      setting.ProfileEnabled,
 		NewsFeedEnabled:                     setting.NewsFeedEnabled,
 		QueryHistoryEnabled:                 hs.Cfg.QueryHistoryEnabled,
+		KioskMode:                           setting.KioskMode,
```
Comment (dhodgsonintergral) [inline pkg/api/frontendsettings.go:132]: settings fro kiosk mode

File: pkg/setting/setting.go [hunk 10]
```diff
@@ -1156,6 +1159,9 @@ func (cfg *Cfg) Load(args CommandLineArgs) error {
 	explore := iniFile.Section("explore")
 	ExploreEnabled = explore.Key("enabled").MustBool(true)
 
+	kiosk := iniFile.Section("kiosk")
```
Comment (dhodgsonintergral) [inline pkg/setting/setting.go:1162]: decides how the options for the grafana.ini, and what key words are required

File: public/app/core/components/AppChrome/AppChrome.test.tsx [hunk 11]
```diff
@@ -67,7 +67,7 @@ const setup = (children: ReactNode) => {
   const renderResult = render(
     <KBarProvider>
       <TestProvider grafanaContext={context}>
-        <AppChrome>
+        <AppChrome hideSearchBar={hideSearch}>
```
Comment (dhodgsonintergral) [inline public/app/core/components/AppChrome/AppChrome.test.tsx:70]: hide search bar

File: public/app/core/components/AppChrome/AppChrome.tsx
Comment (dhodgsonintergral) [inline public/app/core/components/AppChrome/AppChrome.tsx:1]: hide search bar

File: public/app/core/components/AppChrome/AppChromeService.tsx
Comment (dhodgsonintergral) [inline public/app/core/components/AppChrome/AppChromeService.tsx:1]: add embed mode to the options for to work in the via the URL

File: public/app/core/components/AppChrome/NavToolbar/NavToolbar.tsx
Comment (dhodgsonintergral) [inline public/app/core/components/AppChrome/NavToolbar/NavToolbar.tsx:1]: remove the navigation bar

File: public/app/core/components/Footer/Footer.tsx
Comment (dhodgsonintergral) [inline public/app/core/components/Footer/Footer.tsx:1]: remove footer

File: public/app/core/navigation/kiosk.ts
Comment (dhodgsonintergral) [inline public/app/core/navigation/kiosk.ts:1]: allows the kioskmode configuration in the grafana.ini to work

File: public/app/features/commandPalette/actions/staticActions.ts [hunk 12]
```diff
@@ -86,5 +86,5 @@ export default (navBarTree: NavModelItem[]): CommandPaletteAction[] => {
 
   const navBarActions = navTreeToActions(navBarTree);
 
-  return [...globalActions, ...navBarActions];
+  return [...navBarActions];
```
Comment (dhodgsonintergral) [inline public/app/features/commandPalette/actions/staticActions.ts:89]: remove the global actions from the search bar (still accessible with ctrl+k)

File: public/app/features/dashboard/components/DashNav/DashNav.tsx [hunk 13]
```diff
@@ -169,7 +169,7 @@ export const DashNav = React.memo<Props>((props) => {
     const { canStar, canShare, isStarred } = dashboard.meta;
     const buttons: ReactNode[] = [];
 
-    if (kioskMode || isPlaylistRunning()) {
+    if ((kioskMode && kioskMode !== KioskMode.Embed) || isPlaylistRunning()) {
```
Comment (dhodgsonintergral) [inline public/app/features/dashboard/components/DashNav/DashNav.tsx:172]: allows the dashboard configuration items such as favourite to load while in embed kiosk mode

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 14]
```diff
@@ -321,9 +321,9 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
     }
 
     const inspectPanel = this.getInspectPanel();
-    const showSubMenu = !editPanel && !kioskMode && !this.props.queryParams.editview;
+    const showSubMenu = !editPanel && (!kioskMode || kioskMode === KioskMode.Embed) && !this.props.queryParams.editview;
```
Comment (dhodgsonintergral) [inline public/app/features/dashboard/containers/DashboardPage.tsx:324]: allows SubMenu to show in embed kioskmode

File: public/app/features/explore/TraceView/components/TracePageHeader/Actions/TracePageActions.tsx
Comment (dhodgsonintergral) [inline public/app/features/explore/TraceView/components/TracePageHeader/Actions/TracePageActions.tsx:1]: removes request to give grafana feedback

File: public/app/features/templating/template_srv.ts [hunk 15]
```diff
@@ -145,7 +145,7 @@ export class TemplateSrv implements BaseTemplateSrv {
     for (const variable of this.getAdHocVariables()) {
       const variableUid = variable.datasource?.uid;
 
-      if (variableUid === ds.uid) {
+      if (variableUid === ds.uid || variableUid === ds.name) {
```
Comment (dhodgsonintergral) [inline public/app/features/templating/template_srv.ts:148]: allows datasources to be refernces by name when applying adhoc variables

File: public/app/plugins/datasource/grafana/components/QueryEditor.tsx
Comment (dhodgsonintergral) [inline public/app/plugins/datasource/grafana/components/QueryEditor.tsx:1]: remove built in datasources, does not function when all are removed therefore live Measurement in

File: public/app/types/dashboard.ts [hunk 16]
```diff
@@ -96,6 +96,7 @@ export interface DashboardInitError {
 export enum KioskMode {
   TV = 'tv',
   Full = 'full',
+  Embed = 'embed',
```
Comment (dhodgsonintergral) [inline public/app/types/dashboard.ts:99]: map embed for kioskmode

**Review by dhodgsonintergral — COMMENTED** [review]
Comments Explain each part of the code from me.

File: public/app/features/templating/template_srv.ts (hunk 15 above)
Comment (Umaaz) [reply public/app/features/templating/template_srv.ts:148]: look at submitting PR for this fix

File: public/app/intergral/useOpspilotMetadata.ts [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+export const useOpspilotMetadata = () => {
+    useEffect(() => {
+        const event = async (event: MessageEvent) => {
+          if (event.data.type === 'opspilot-host.getMetadata') {
+            window.parent.postMessage({type: "opspilot-slave.sendMetadata", metadata: { 
+              slaveUrl: window.location.pathname,
+              timeStart: getTimeSrv().timeRange().from.valueOf(), 
+              timeEnd: getTimeSrv().timeRange().to.valueOf(),
+              timezone: getTimeSrv().timeModel?.getTimezone() === 'browser' ? Intl.DateTimeFormat().resolvedOptions().timeZone : getTimeSrv().timeModel?.getTimezone(),
+            } }, '*');
```
Comment (Umaaz) [inline public/app/intergral/useOpspilotMetadata.ts:15]: possible security issue with wildcard origin allowance.

**Review by Nihasa350 — APPROVED** [review]

**Pull request merged**